test/LocatedTriplesTest.cpp

```diff
@@ -171,3 +171,197 @@ TEST_F(LocatedTriplesTest, mergeTriples) {
+          blockIndices.push_back(blockIndex);
+        }
+        std::sort(blockIndices.begin(), blockIndices.end());
+        for (auto blockIndex : blockIndices) {
+          ASSERT_TRUE(locatedTriplesPerBlock.map_.contains(blockIndex))
+              << "blockIndex = " << blockIndex << " not found";
+          std::ostringstream os;
+          os << locatedTriplesPerBlock.map_.at(blockIndex);
+          std::string computedLocatedTriples = os.str();
+          std::string expectedLocatedTriples =
+              expectedLocatedTriplesPerBlock.at(blockIndex);
+          ASSERT_EQ(computedLocatedTriples, expectedLocatedTriples)
+              << "blockIndex = " << blockIndex;
+        }
+        ASSERT_EQ(locatedTriplesPerBlock.map_.size(),
+                  expectedLocatedTriplesPerBlock.size());
+
+        // Delete the permutation files.
+        ad_utility::deleteFile(permutationFilename);
+        ad_utility::deleteFile(permutationFilename + MMAP_FILE_SUFFIX);
+      };
+
+  // Triples in the index.
+  IdTable triplesInIndex = makeIdTableFromVector({{1, 10, 10},    // Row 0
+                                                  {2, 10, 10},    // Row 1
+                                                  {2, 15, 20},    // Row 2
+                                                  {2, 15, 30},    // Row 3
+                                                  {2, 20, 10},    // Row 4
+                                                  {2, 30, 20},    // Row 5
+                                                  {2, 30, 30},    // Row 6
+                                                  {3, 10, 10}});  // Row 7
+
+  // Locate the following triples, some of which exist in the relation and some
+  // of which do not, and which cover a variety of positons, including triples
+  // that are larger than all existing triples.
+  IdTable triplesToLocate =
+      makeIdTableFromVector({{2, 15, 20},    // Equals Row 2
+                             {2, 14, 20},    // Before Row 2
+                             {2, 20, 10},    // Equals Row 4
+                             {2, 30, 20},    // Equals Row 5
+                             {2, 30, 30},    // Equals Row 6
+                             {2, 30, 31},    // Before Row 7
+                             {9, 30, 32}});  // Larger than all.
+
+  // Now test for multiple block sizes (16 bytes is the minimum).
+  // testing::internal::CaptureStdout();
+  std::cout << "Index triples: " << triplesInIndex << std::endl;
+  std::cout << "Delta triples: " << triplesToLocate << std::endl;
+
+  // With block size 16, we have each triple in its own block.
+  testWithGivenBlockSize(
+      triplesInIndex, triplesToLocate, 16,
+      {{2, "{ LT(2 0 V:2 V:14 V:20 0) LT(2 0 V:2 V:15 V:20 1) }"},
+       {4, "{ LT(4 0 V:2 V:20 V:10 1) }"},
+       {5, "{ LT(5 0 V:2 V:30 V:20 1) }"},
+       {6, "{ LT(6 0 V:2 V:30 V:30 1) }"},
+       {7, "{ LT(7 0 V:2 V:30 V:31 0) }"},
+       {8, "{ LT(8 x V:9 V:30 V:32 0) }"}});
```

Comment:
I like the structure of these tests in general. I would, as already stated, strictly prefer if there were proper objects for testing and not strings, e.g.
```
using LT = LocatedTriple;
LocatedTriplesMap expected {{2, {LT(2, 0, V(2), V(14), V(20, 0, LT{2, 0, V(2), V(15), V(20}}, ...}....
```
You can make this work in a typesafe way with a syntax that is still rather readable.